### PR TITLE
Disjunction: Fix autoloading

### PR DESCRIPTION
**Pull request by vonglasow**
Need to find another way to fix it.

### CODE REVIEW

File: Model/Specification.php [hunk 1]
```diff
@@ -39,7 +39,8 @@
 use Hoa\Consistency;
 
 if (false === Consistency::entityExists('Hoa\Realdom\Disjunction')) {
-    Consistency\Autoloader::load('Hoa\Realdom\Disjunction');
+    $autoloader = new Consistency\Autoloader();
+    $autoloader->load('Hoa\Realdom\Disjunction');
```
Comment (Hywan) [inline Model/Specification.php]: The namespace directory is not specified. How you sure it will work?

**Comment by vonglasow** [comment]
I am sure this will not work at this state. As I have seen we need to add namespace directory but I don't know what to do about the latest line regarding the class Hoa\Consistency\Autoloader

```php
/**
 * Autoloader.
 */
$autoloader = new Autoloader();
$autoloader->addNamespace('Hoa', dirname(__DIR__));
$autoloader->register();
```

I can add namespace directory for this 3 call but which syntax used (php7 implements new syntax for dirname)

**Comment by vonglasow** [comment]
Looks ok for autoloading. @Hywan you can review it.

**Comment by vonglasow** [comment]
@Hywan Ok so if we assume than realdom must exists. We don't need to load it in this way. But we must found a way to load it? Do we assume also than composer will load it ?

**Comment by Hywan** [comment]
@vonglasow I guess the best way to do so is…
1. Autoload the `Hoa/Realdom/Disjunction.php` file.
2. Remove the autoloading here.

**But**, there is no particular reason to load `Hoa/Realdom/Disjunction.php` for every requests. So, instead of calling the autoloader manually, we can call `entityExists` with its second parameter set to `true` (to load the autoloader in case it does not exist). And if it still does not exist, throw an `RuntimeException` or a `Hoa\Praspel\Exception`. Why? Because we have no idea where `Hoa\Realdom` is located. We can guess it but we are likely to be wrong.

So, here is my proposal:
```php
if (false === Consistency::entityExists('Hoa\Realdom\Disjunction', true)) {
    throw new Praspel\Exception('Hoa\Realdom seems to not be loaded.');
}
```
This case will never happen because `hoa/praspel` has a dependency to `hoa/realdom`, so `entityExists` will always load `Hoa\Realdom\Disjunction` without any problem.

Thoughts?

File: Model/Specification.php [hunk 2]
```diff
@@ -38,8 +38,8 @@
 
 use Hoa\Consistency;
 
-if (false === Consistency::entityExists('Hoa\Realdom\Disjunction')) {
-    Consistency\Autoloader::load('Hoa\Realdom\Disjunction');
+if (false === Consistency::entityExists('Hoa\Realdom\Disjunction', true)) {
+    throw new Praspel\Exception('Hoa\Realdom seems to not be loaded.');
```
Comment (Hywan) [inline Model/Specification.php]: Maybe `use Hoa\Praspel` is missing?

**Comment by Hywan** [comment]
Fixed and merged in https://github.com/hoaproject/Praspel/commit/d42a5c204a9fda83ac9570c56361cc3e0f35bd06.

Thanks!

A new snapshot has been released since this is a bug fix: https://github.com/hoaproject/Praspel/releases/tag/1.16.05.09.

**Pull request closed without merging**